### ISSUE TITLE
Cart/checkout block - Third party plugin compatibility - Need a replacement for hook woocommerce_cart_item_visible

### ISSUE THREAD

**Issue by marcusig** [woocommerce/woocommerce#42269]
## Is your feature request related to a problem? Please describe.
I have a Product Configurator plugin which has the option to add linked product to the cart.
Those products are linked to a parent product, and the shop owner has the option to hide those items in the cart. 
These items are required by the admin for order management, but are not useful to the customer (e.g. various parts of a product).

## Describe the solution you'd like
In the cart shortcode, this is achieved using the filter `woocommerce_cart_item_visible`. 
My suggestion would be to have that same hook in the cart and checkout blocks, but you may have a better solution.
CSS isn't an option, and I believe the item still needs to be present in the API data.

## Additional context

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11990914/a3a8a42a-ab59-4dae-b452-84cbec94b1d0)

vs block:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/11990914/3aa5ed35-e461-4cc6-82ee-fb0fc0827d48)

**Comment by nielslange** [comment]
Thanks for reaching out, @marcusig. Looking at the screenshot of the classic cart, I wonder if hiding sub products, that contain a price, might lead to legal problems. The screenshot shows that the cart line item costs €1559.00, while the total price is €1624.00. Looking at the classic cart, it's not clear why the total price is €65.00 is higher than the cart line item price.

**Comment by marcusig** [comment]
@nielslange I wouldn't worry about legal questions here, as this is a fictitious example. 
In practice, the 65 would be included in the parent product as one of the options. 

Take for example a bicycle configurator, where the customer only sees one product, the bicycle. 
The bicycle is made of a list of parts, such as the wheels, pedals, gears... 
The shop also sells those parts separately, as spares, and manages the stock of all those in one place. 
When a customer places a bicycle order, they only need to see the main product. 
But the shop owner needs to have all the parts attached to that order, mainly for exporting and using in third party software.

It's a bit like a composite product if you wish.

This is a feature which is already in use by many of my customers.

**Comment by nielslange** [comment]
Thanks for your prompt reply, @marcusig. I also checked for other extensions that could benefit from a `woocommerce_cart_item_visible` when using WooCommerce Blocks:

<table>
<tr>
<td valign="top">

### [Product Bundles](https://woocommerce.com/products/product-bundles/)
<br><br>
<img width="552" alt="Screenshot 2023-08-11 at 15 29 48" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/23083005-1b78-4c18-ba2a-99a79413736a">
</td>
<td valign="top">

### [Composite Products](https://woocommerce.com/products/composite-products/)
<br><br>
<img width="551" alt="Screenshot 2023-08-11 at 15 34 01" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/45becd05-03d1-41ed-9182-ebfe649074ad">
</td>
<td valign="top">

### [Mix and Match Products](https://woocommerce.com/products/woocommerce-mix-and-match-products/)
<br><br>
<img width="560" alt="Screenshot 2023-08-11 at 15 41 50" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/1499cf2f-f09a-433b-845c-4c7dccbeab28">
</td>
</tr>
</table>

I'll move this issue to our backlog for now.

**Comment by pierorocca** [comment]
@yjailin this makes sense to me that optionally a product plugin would be able to hide child products, particularly for this configure-to-order use case. I don't see a problem with it and wanted to run it by you.

**Comment by yjailin** [comment]
@pierorocca Makes sense to me for configure-to-order. One condition though: the hidden child's price has to be included in the parent's displayed price.

Look at the first screenshot. The line item says €1559 but the total is €1624, and nothing on the page explains where the €65 came from. That's the case to avoid. As long as the parent price covers the children, hiding them is fine.
